src/Flysystem/FlysystemAssetStore.php

```diff
@@ -1071,22 +1069,22 @@ protected function getStreamSHA1($stream)
      *
      * @param resource $stream
      * @return string Filename of resulting stream content
-     * @throws FlysystemException
+     * @throws Exception
      */
     protected function getStreamAsFile($stream)
     {
         // Get temporary file and name
         $file = tempnam(sys_get_temp_dir(), 'ssflysystem');
         $buffer = fopen($file ?? '', 'w');
         if (!$buffer) {
-            throw new FlysystemException("Could not create temporary file");
+            throw new Exception("Could not create temporary file");
```

Comment:
Why was this changed? ([commit](https://github.com/silverstripe/silverstripe-assets/compare/11d4da5..88b62be)) It wasn't causing any errors in CI? If there isn't a compelling reason for changing this can we please revert this change?

Reply:
`League\Flysystem\Exception` class doesn't exist anymore in new version of Flysystem.

Reply:
Perhaps this should throw `League\Flysystem\UnableToWriteFile`?

Reply:
I'm kind on inclined to not use `UnableToWriteFile` here, mainly because that's thrown exclusively by league code, makes sense to me to no use the league exception for exception in our own logic. I don't think that errors with writing to temporary files really warrants a new custom Exception, so plain old Exception is fine IMO

Reply:
We're catching and rethrowing it from our adapter, but fair enough as it's not really related to the actual file manipulations. Actually made me look up flysystem's exceptions so some good still came out of the discussion.